docs/source/how_to_add_a_component.rst

```diff
@@ -0,0 +1,112 @@
+How to Add a Component
+======================
```

Comment:
I think the upper part of this documentation would be confusing to someone not familiar with the code and "dashboard terminology". The lower part where you go over the steps is great! :+1: 

I would suggest the following to make things clearer:

1. Define what a component is.
2. Make sure you always use the same terms / words to refer to the same steps and constructs.
3. Start with a short description of the steps and main modules (`from_data_to_dashboard.py` and `run_dashboard.py`).
4. Maybe take the simplest of our present components and show for each step what would be added to add the component as a whole (might be too time consuming to do this).